### ISSUE TITLE
HDMI port not working

### ISSUE THREAD

**Issue by kitaliev** [daliansky/XiaoMi-Pro-Hackintosh#722]
XiaoMi Notebook Pro 2019 MX250 i7-8550U KBL

Sorry don't know how to find EFI version, but OpenCore is 0.7.7

External monitor is not working through HDMI port, I double checked everything
Works perfectly on this laptop on windows which is installed on second ssd

[EFI.zip](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/files/11240538/EFI.zip)
[system.log](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/files/11240539/system.log)

**Comment by stevezhengshiqi** [comment]
You can try re-plug it, or close the lid for about five seconds and reopen the lid.

**Comment by kitaliev** [comment]
> You can try re-plug it, or close the lid for about five seconds and reopen the lid.

Yeah, i know and i was trying it about a year. I used hackintosh on my laptop for about a year and a was aware of this problem but i didn't have a need for external monitor until now, and when i needed to work with external monitor i just used windows but now i need both, hackintosh and external monitor, so i reinstalled it to update everything thinking maybe my problem was solved in some of updates but no

**Comment by stevezhengshiqi** [comment]
Understand the difficulty. If that's the case, I recommend to use USB-C DP port to connect with external monitors, which is more stable.

**Comment by kitaliev** [comment]
Im not throwing it up on community i just dont have an experience with setting up hackintosh and fixing it's stuff so i will be satisfied with some advice where to dig and what to read to fix it. Naturally i read dortanias opencore guide from A to Z and know how to make patches and changes.

**Comment by kitaliev** [comment]
> Understand the difficulty. If that's the case, I recommend to use USB-C DP port to connect with external monitors, which is more stable.

Will it work if i just plug it in or i need to make some changes to config.plist?

**Comment by kitaliev** [comment]
Anyway please leave this issue open i will continue to look for a solution and will write about what i found here. Or maybe someone else will find something useful

**Comment by stevezhengshiqi** [comment]
> Will it work if i just plug it in or i need to make some changes to config.plist?

It will work out of box for most cases.

Theoretically, both HDMI and DP work with no problem, at least on my laptop. But based on GitHub user reports, macOS HDMI reliability is a mystery, relates to resolution setting, monitor brand, etc. We have tested and injected every possible property, and it works on some laptops.

In your case, I recommend to use DP port as it's the most effective way to connect to external monitors. I have limited capability to debug HDMI problem, because I never met any HDMI issue. Feel free to come up with ideas and test it. We can disucss more if it works on you.

![IMG_2490](https://user-images.githubusercontent.com/36894554/232326275-d2fbf205-0091-4f31-ba30-9d032c13eda2.jpeg)


Similiar problem: https://github.com/daliansky/XiaoMi-Pro-Hackintosh/issues/117

**Comment by dongmuhds** [comment]
I also encountered this issue in the latest version, but in previous versions, it was possible to plug and unplug the HDMI, and even if it was plugged and unplugged, the latest version was invalid.

**Comment by stevezhengshiqi** [comment]
> I also encountered this issue in the latest version, but in previous versions, it was possible to plug and unplug the HDMI, and even if it was plugged and unplugged, the latest version was invalid.

Interesting. If you have chance, could you test which specific release causes the issue? It happens on release v1.7.7 or v1.7.6?

**Comment by dongmuhds** [comment]
You're right, I think it happened in version 1.77

**Comment by dongmuhds** [comment]
Yes, I am not in a hurry to use the HDMI interface and wait for the next version update to see if it can solve this problem.

**Comment by dongmuhds** [comment]
经过我的测试，发现了问题，是因为我改变了仿冒苹果的机型。我在远景论坛上找到了相关讨论，经过测试14,1  15,2 15,4可以正常使用核显hdmi显示。我将15,3改为了15,2后正常驱动了hdmi显示输出。
After my testing, I found the problem because I changed the model that imitated Apple. I found relevant discussions on the Pcbeta Forum, and after testing 14,1 15,2 15,4 can use the core display HDMI display normally. I changed 15,3 to 15,2 and then drove the HDMI display output normally.
远景论坛(Pcbeta Forum)地址：https://bbs.pcbeta.com/viewthread-1849120-1-1.html

**Comment by stevezhengshiqi** [comment]
@dongmuhds 更改机型时建议不要修改15寸MacBookPro机型，例如MacBookPro15,3. 这些机型是双显卡且基本配有标压u，和小米本配置差别太大。
